apps/prairielearn/src/lib/question-variant.ts

```diff
@@ -214,10 +242,10 @@ async function makeAndInsertVariant(
       // generating two variants for the same instance question. If we're the
       // second one to try and insert a variant, just pull the existing variant
       // back out of the database and use that instead.
-      const existing_variant = await sqldb.callOptionalRow(
-        'instance_questions_select_variant',
-        [instance_question_id, require_open],
-        VariantWithFormattedDateSchema.nullable(),
+      const existing_variant = await selectVariantForInstanceQuestion(
+        instance_question_id,
+        require_open,
+        true, // lock acquired
```

Comment:
Good point. I thought of that briefly, since the impact of that would be that the call in ensureVariant (line 356) would be done without locking the assessment instance. Now that I think of it properly, any race conditions caused by that code not locking the assessment instance would not be prevented by locking the assessment instance itself, so I'm ok with getting rid of the lock. Removed in 96544360eda9ca69b5b3e990cb3455876a0233ba.